src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -100,6 +101,43 @@ protected ImmutableArray<ISymbol> GetSymbols(IParameterSymbol parameter, int pos
                 .ToImmutableArray();
         }
 
+        private ImmutableArray<ITypeSymbol> SubstituteTypeParameters(ImmutableArray<ITypeSymbol> parameterTypeSymbols, SyntaxNode invocationExpression)
+        {
+            if (!parameterTypeSymbols.Any(t => t.IsKind(SymbolKind.TypeParameter)))
+            {
+                return parameterTypeSymbols;
+            }
+
+            var invocationSymbols = _context.SemanticModel.GetSymbolInfo(invocationExpression).GetAllSymbols();
+            if (invocationSymbols.Length == 0)
+            {
+                return parameterTypeSymbols;
+            }
+
+            var concreteTypes = ArrayBuilder<ITypeSymbol>.GetInstance();
+            foreach (var invocationSymbol in invocationSymbols)
+            {
+                var typeParameters = invocationSymbol.GetTypeParameters();
+                var typeArguments = invocationSymbol.GetTypeArguments();
+
+                foreach (var parameterTypeSymbol in parameterTypeSymbols)
+                {
+                    if (parameterTypeSymbol.IsKind<ITypeParameterSymbol>(SymbolKind.TypeParameter, out var typeParameter))
+                    {
+                        var index = typeParameters.IndexOf(typeParameter);
+                        var concreteType = typeArguments.ElementAtOrDefault(index);
+                        concreteTypes.Add(concreteType ?? typeParameter);
```

Comment:
not sure ths is correct.  i think you don't want to do anything if the type parameter is not int he list of typeParmeters right?

Reply:
Need to leave it there because
1) It could be the only type, and we need it later to pull in members of object
2) It could have a type constraint from which to pull members

I've added tests for both of these.